backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -355,22 +355,100 @@ CopyPropagationAndElimination::copyPropAndDeadCodeElim(
         } else if (auto jc = stmt->to<IR::DpdkJmpLessStatement>()) {
             instr.push_back(new IR::DpdkJmpLessStatement(jc->label,
                         replaceIfCopy(jc->src1, false), replaceIfCopy(jc->src2)));
+        } else if (auto jc = stmt->to<IR::DpdkJmpLessOrEqualStatement>()) {
```

Comment:
Just curious. for safe programming, shouldn't we be using is<TYPE> before converting to to<TYPE>?

Reply:
I think `is` is for when we do not really need the casting result but just the check., But when we need the casting result doing `is` first is unnecessary overhead.
Also this kind of style is prevalent in p4c source everywhere i.e. https://github.com/p4lang/p4c/blob/main/frontends/p4/typeMap.cpp#L23

Reply:
The to<>() result is assigned to var, which is checked in if condition. This is equivalent to 
if (is<>())
  auto var = to<>()